File: GeneratedEntityReaderClasses/Components/TeleportProjectileComponent.cs
namespace GeneratedNoitaClasses;
public class TeleportProjectileComponent : NoitaComponentBase,  iNoitaType<TeleportProjectileComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaFloat min_distance_from_wall { get; set; } = new();
	public NoitaInt actionable_lifetime { get; set; } = new();
	public NoitaBool reset_shooter_y_vel { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		min_distance_from_wall.Read(s);
		actionable_lifetime.Read(s);
		reset_shooter_y_vel.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		min_distance_from_wall.Write(s);
		actionable_lifetime.Write(s);
		reset_shooter_y_vel.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/TeleportProjectileComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaFloat
from BasicNoitaTypes import NoitaInt

from NoitaStream import NoitaStream
@dataclass
class TeleportProjectileComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    min_distance_from_wall:NoitaFloat = field(default_factory=NoitaFloat)
    actionable_lifetime:NoitaInt = field(default_factory=NoitaInt)
    reset_shooter_y_vel:NoitaBool = field(default_factory=NoitaBool)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.min_distance_from_wall.Read(s)
        self.actionable_lifetime.Read(s)
        self.reset_shooter_y_vel.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.min_distance_from_wall.Write(s)
        self.actionable_lifetime.Write(s)
        self.reset_shooter_y_vel.Write(s)

